bytelatent/model/blt.py

```diff
@@ -767,7 +768,10 @@ def compute_hash_embeddings(
     return local_encoder_embeds
 
 
-class ByteLatentTransformer(nn.Module, SequenceModelWithOutput):
+class ByteLatentTransformer(nn.Module, SequenceModelWithOutput, PyTorchModelHubMixin,
+                            repo_url="https://github.com/facebookresearch/blt",
+                            pipeline_tag="text-generation",
+                            license="cc-by-nc-sa-4.0"):
```

Comment:
Just to confirm, this is referring to the code having this license, not the weights? The code is under this, but the weights are under the fair research license.

Reply:
Ok thanks, pushed a commit to make it license="other". You can pass any license present here: https://huggingface.co/docs/hub/en/repositories-licenses

Reply:
The license gets pushed to the model repo's metadata, so it refers to the license of the weights.

Reply:
Mentioned in another comment, but the most important things for us on metadata are to make sure that:
-  On the model card/readme.md https://huggingface.co/facebook/blt/blob/main/README.md, `license: other` and `extra_gated_fields` stays the same. I'll look at the mixin options, but if it doesn't overwrite existing settings, then that would work.
- In the HF repository, the LICENSE file is not overwritten, since `license: other` refers to that license in our case.

Reply:
Feel free to push to a separate model repository first to try it out and be safe :) you can definitely still add `extra_gated_fields` after pushing. I would recommend just copying the current model card after pushing (and adding the extra `pipeline_tag: text-generation`).